sdk/tests/conformance/glsl/bugs/constant-precision-qualifier.html

```diff
@@ -67,6 +67,30 @@
     gl_FragColor = vec4(0.0, 2.0 * a, 0.0, 1.0);
 }
 </script>
+<script id="fshaderNoConstants" type="x-shader/x-fragment">
+// This shader has the same functionality as the one above, but it doesn't contain
+// operations that can be constant folded at compile-time.
+// It's here to provide a point of comparison.
+uniform mediump float uTest;
+
+void main() {
+    highp float c = 4096.5 + uTest;
+    mediump float a = 0.0;
+    a = fract(c + uTest);
+    gl_FragColor = vec4(0.0, 2.0 * a, 0.0, 1.0);
+}
+</script> <script id="fshaderAllHighp" type="x-shader/x-fragment">
+// This shader has the same functionality as the one above, but it only uses highp.
+// It's here to provide a point of comparison.
+uniform highp float uTest;
```

Comment:
highp is optional for fragment shaders in GLSL 1. You should guard it using GL_FRAGMENT_PRECISION_HIGH.